File: Difficulty: Medium/Subset Sums/subset-sums.cpp
class Solution {
  public:
    void solve(int idx,vector<int> &arr,vector<int> &ans,int sums){
        if(idx==arr.size()){
            ans.push_back(sums);
            return;
        }
        
        solve(idx+1,arr,ans,sums+arr[idx]);
        solve(idx+1,arr,ans,sums);
        
    }
  
    vector<int> subsetSums(vector<int>& arr) {
        // code
        vector<int> ans;
        int sums=0;
        solve(0,arr,ans,sums);
        sort(ans.begin(),ans.end());
        return ans;
        
    }
};

File: Difficulty: Medium/Subset Sums/subset-sums.java
// User function Template for Java//User function Template for Java
class Solution {
    void solve(int idx,int[] arr,int sub,ArrayList<Integer> ans){
        if(idx==arr.length){
            ans.add(sub);
            return;
        }
        
        sub+=arr[idx];
        solve(idx+1,arr,sub,ans);
        sub-=arr[idx];
        solve(idx+1,arr,sub,ans);
    }
    
    public ArrayList<Integer> subsetSums(int[] arr) {
        // code here
        ArrayList<Integer> ans = new ArrayList<>();
        int sub=0;
        
        solve(0,arr,sub,ans);
        
        return ans;
        
    }
}